content/blog/registry-writing-i2c-component/index.md

```diff
@@ -0,0 +1,742 @@
+
+To create the component, we will go through some steps:
+
+1. Create the new component using the command line.
+2. Create the component manifest file and set with the component information.
+3. Set the license.
+4. Write the README.md file.
+5. Write the code for the sensor driver.
+6. Write the example.
+
+We assume that you already have the ESP-IDF installed on your system. If you don't, please check our article about installing the ESP-IDF.
+
+On the Espressif DevCon23, Ivan Grokhotkov, gave a talk titled: Developing, Publishing, and Maintaining Components for ESP-IDF. You can watch this talk as an additional material for your studies.
+
+{{< youtube D86gQ4knUnc >}}
+
+#### Create the new component
+
+The process will be done by the CLI (Command-line Interface) tool `idf.py`.
+
+```bash
+idf.py create-component shtc3
+```
+
+After that, the new folder should contain the following structure:
+
+```text
+.
+└── shtc3
+    ├── CMakeLists.txt
+    ├── include
+    │   └── shtc3.h
+    └── shtc3.c
+```
+
+#### Create the Manifest file
+
+The manifest file is mandatory and this is the file that the Registry will recognize it as a component. The bare-minimum for this file is described in the documentation, but we will fill with the most common important fields.
+
+Create the `idf_component.yml` in the component root directory with this content. You can change it if you want.
+
+```yaml
+version: 1.0.0
+targets:
+  - esp32
+  - esp32s2
+  - esp32s3
+  - esp32c3
+  - esp32c2
+  - esp32c6
+  - esp32h2
+  - esp32p4
+description: SHTC3 Temperature and humidity sensor driver for ESP-IDF
+url:
+repository:
+issues:
+maintainers:
+  - "Author Name <author@email.com>"
+tags:
+  - shtc3
+  - sensor
+  - temperature
+  - humidity
+  - i2c
+  - driver
+dependencies:
+  idf : ">=5.3"
+```
+
+- **version**: This contains the component version following the version format `major.minor.patch` (0.0.0).
+- **targets**: This is the list of the supported targets.
+- **description**: Brief description about the component.
+- **url**: URL to the component page or company page.
+- **repository**: URL to the GitHub component repository ending with `.git`.
+- **issues**: URL to the place you can open issues for this component.
+- **maintainers**: List of the maintainers with email.
+- **tags**: Tags that might be relevant for searching in the Registry page.
+- **dependencies**: You can add the dependencies, like the ESP-IDF minimum version or other components.
+
+This component will require the ESP-IDF v5.3 or higher (latest stable release in Oct 2024).
+
+#### Set the License
+
+Now we need to define the license file. This is a very important step for any open-source project.

```

Comment:
Could you provide a link to recommended licenses? Like MIT or Apache? So that people know which LICENSE template they can use or where can they take it from.